schema.cc

```diff
@@ -656,6 +658,145 @@ std::ostream& operator<<(std::ostream& os, const schema& s) {
     return os;
 }
 
+std::ostream& map_as_cql_param(std::ostream& os, const std::map<sstring, sstring>& map, bool first = true) {
+    for (auto i: map) {
+        if (first) {
+            first = false;
+        } else {
+            os << ",";
+        }
+        os << "'" << i.first << "': '" << i.second << "'";
+    }
+
+    return os;
+}
+
+std::ostream& column_definition_as_cql_key(std::ostream& os, const column_definition & cd) {
+    os << cd.name_as_cql_string();
+    os << " " << cd.type->cql3_type_name();
+
+    if (cd.kind == column_kind::static_column) {
+        os << " STATIC";
+    }
+    return os;
+}
+
+static sstring remove_index_suffix(const sstring& name) {
+    return name.substr(0, name.size() - 6); // remove the _index suffix from an index name;
+}
+
+static bool is_secondary_index(const utils::UUID& id, const schema& s) {
+    return  service::get_local_storage_service().db().local().find_column_family(id).get_index_manager().is_global_index(s);
+}
+std::ostream& schema::describe(std::ostream& os) const {
+    os << "CREATE ";
+    int n = 0;
+
+    if (is_view()) {
+        if (is_secondary_index(view_info()->base_id(), *this)) {
+            os << "INDEX " << remove_index_suffix(cf_name()) << " ON " << ks_name() << "." << view_info()->base_name() << "(";
+            for (auto& pk : partition_key_columns()) {
```

Comment:
I'll add one

Reply:
Did you? You have to. Otherwise, your code will seem to work until someone happens to use it with local secondary indexes, and things just silently break :-(
We *can* commit code without the local-secondary-index part and fix this bug later, hoping nobody uses local secondary indexes (?) but that's not a technical decision but rather a decision on how quickly we need a almost-working fix in. I have no input on that. Maybe @slivne does?